animatplot/animation.py

```diff
@@ -144,8 +146,11 @@ def timeline_slider(self, text='Time', ax=None, valfmt='%1.2f', color=None,
 
         def set_time(t):
             self.timeline.index = int(self.slider.val)
-            self.slider.valtext.set_text(
-                self.slider.valfmt % (self.timeline[self.timeline.index]))
+            if np.issubdtype(self.timeline.t.dtype, np.datetime64):
```

Comment:
Changing the default arg of ```valfmt``` to ```None```, and then first checking if the valfmt is not None, then checking if the instance is a np.datatime64 can setting valfmt to ```%s``` if that is the case, and ```%1.2f``` in all other cases. This way if a user passes in a value it till not be overridden